spec/controllers/secret_shares_controller_spec.rb

```diff
@@ -0,0 +1,14 @@
+describe SecretSharesController do
+  describe "POST create" do
+    it "Creates Secret Share Identifier" do
+      new_user = create(:user1)
+      sign_in new_user
+      new_category = create(:category, userid: new_user.id)
+      new_mood = create(:mood, userid: new_user.id)
+      new_strategies = create(:strategy, userid: new_user.id)
+      new_moment = create(:moment, userid: new_user.id, category: Array.new(1, new_category.id), mood: Array.new(1, new_mood.id), strategies: Array.new(1, new_strategies.id))
```

Comment:
I'd love to have a chat about this tomorrow but there are a couple of things I'd like to point out now.

It doesn't make sense to use `Array.new` if you are only creating 1 item, just use `[]` like `[new_category.id]`. I've been removing any instances of these I find in the Rails 5 upgrade.

Since the keys in the creation of a moment are singular, it tells me you don't actually need them to be arrays, you can just:
```
category: new_category,
mood: new_mood,
strategies: new_strategies
```

Saying all this, I don't think you need to create all those objects. Using `create` actually stores a record in the database so, for the sake of test performance, you only want to do it when you really need to.
I believe you only need to create a user and a moment.